### ISSUE TITLE
The SSE2 detection fails on i386

### ISSUE THREAD

**Issue by yurivict** [vxl/vxl#761]
```
-- Performing Test VXL_HAS_SSE2_HARDWARE_SUPPORT
-- Performing Try-Run Test VXL_HAS_SSE2_HARDWARE_SUPPORT - Test Compilation Failed
```

http://beefy17.nyi.freebsd.org/data/head-i386-default/p526741_s358237/logs/vxl-2.0.2.629.log (IPv6 URL)

FreeBSD 13 i386

**Comment by hjmjohnson** [comment]
This is a longstanding problem.  Any help in fixing this issue would be greatly appreciated.  It is unlikely to get fixed without someone in the community providing a fix.

**Comment by yurivict** [comment]
Something like
```
if (CMAKE_SYSTEM_PROCESSOR STREQUAL "i386")
  set(CFLAGS ${CXXFLAGS} -msse2)
  set(CXXFLAGS ${CXXFLAGS} -msse2)
endif()
```
should do.

**Comment by hjmjohnson** [comment]
@yurivict I don't think it is a good idea to automatically set values like this. VXL does not 'require' sse2 to work, but if chosen, or specified, it should take advantage of the feature. 

I've made an update to make the implementation a bit cleaner, and now require the developer to request the compiler features desired.

Would you mind seeing if the current master branch solves your issues?  It should now provide better diagnostic messages if an incompatible configuration is selected.

**Comment by yurivict** [comment]
I am not able to build on i386 on demand because of the cost of setting up VMs.
But I will provide feedback in case the next release fails.

**Comment by hjmjohnson** [comment]
Closing as platform untested. i386 (32-bit x86) is not part of the supported CI matrix for vxl 7.0 — all dashboard builds are x86_64 / arm64. The SSE2 try-run failure leaves VXL_HAS_SSE2_HARDWARE_SUPPORT=OFF on i386, which is the safe fallback. If 32-bit support is needed, please open a focused issue identifying the platform/distro and proposed CI coverage.
